src/Fmt_ast.ml

```diff
@@ -1697,7 +1698,12 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
                              | Some i when i <= 2 -> "@ "
                              | _ -> "@;<1 2>" ) )
                        $ fmt_expression c ?box xbody )
-                   $ str ")" $ Cmts.fmt_after c pexp_loc )
+                   $ fmt_or_k
+                       Poly.(c.conf.indicate_multiline_delimiters = `Space)
+                       (fmt_or_k fit (str ")")
+                          (fits_breaks ~force_fit_if:fit ")" ~hint:(1, 0) ")"))
+                       (fits_breaks ~force_fit_if:fit ")" ~hint:(0, 0) ")")
```

Comment:
I don't remember why this parentheses are gone. @gpetiot what do you think ?
Could it be that there was no space after the opening ?
eg, it formats like this:
```
   foo (fun () ->
       ...
       abc )

  foo
     (let a = ... in
      a)
```

Reply:
In such cases the opening space was originally taken to be one following `(fun`.

Reply:
I don't remember either, and I think it was decided it was okay to sometimes have no space at the beginning but to have a space at the end, especially for `(fun` even if `indicate-multiline-delimiters=space`. I didn't manage to find the issue/PR where this have been discussed. There have been a lot of PRs to improve the consistency of parentheses, but maybe some regressions have been introduced along the way. I'm favorable to improve the consistency anyway.

Reply:
I moved that commit to a PR: https://github.com/ocaml-ppx/ocamlformat/pull/1169